ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HiveMaterializedViewUtils.java

```diff
@@ -403,4 +492,28 @@ private static RelNode copyNodeScanNewCluster(RelOptCluster optCluster, RelNode
     }
     return newScan;
   }
+
+  public static MaterializationSnapshot getSnapshotOf(DDLOperationContext context, Set<TableName> tables)
+          throws HiveException {
+    Map<String, String> snapshot = getSnapshotOf(context.getDb(), tables);
+    if (snapshot.isEmpty()) {
+      return new MaterializationSnapshot(context.getConf().get(ValidTxnWriteIdList.VALID_TABLES_WRITEIDS_KEY));
+    }
+
+    return new MaterializationSnapshot(snapshot);
+  }
+
+  private static Map<String, String> getSnapshotOf(Hive db, Set<TableName> tables) throws HiveException {
+    Map<String, String> snapshot = new HashMap<>(tables.size());
+    for (TableName tableName : tables) {
+      Table table = db.getTable(tableName);
+      if (table.getStorageHandler() != null) {
+        String sh = table.getStorageHandler().getCurrentSnapshotId(table);
+        if (isNotBlank(sh)) {
```

Comment:
`sh` will be never empty or null.

Reply:
It depends on the implementation: the default is `null`
HiveStorageHandler.java
```
  default String getCurrentSnapshotId(org.apache.hadoop.hive.ql.metadata.Table table) {
    return null;
  }
```